plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -5,31 +5,58 @@
 
 package org.opensearch.ml.cluster;
 
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_INDEX;
+
+import java.time.Instant;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.Semaphore;
 
 import lombok.extern.log4j.Log4j2;
 
 import org.opensearch.action.ActionListener;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.update.UpdateRequest;
 import org.opensearch.client.Client;
 import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.ml.common.MLModel;
+import org.opensearch.ml.common.model.MLModelState;
 import org.opensearch.ml.common.transport.sync.MLSyncUpAction;
 import org.opensearch.ml.common.transport.sync.MLSyncUpInput;
 import org.opensearch.ml.common.transport.sync.MLSyncUpNodeResponse;
 import org.opensearch.ml.common.transport.sync.MLSyncUpNodesRequest;
+import org.opensearch.ml.indices.MLIndicesHandler;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
 
 @Log4j2
 public class MLSyncUpCron implements Runnable {
 
+    public static final int LOAD_MODEL_TASK_GRACE_TIME_IN_MS = 20_000;
     private Client client;
+    private ClusterService clusterService;
     private DiscoveryNodeHelper nodeHelper;
+    private MLIndicesHandler mlIndicesHandler;
+    @VisibleForTesting
+    Semaphore updateModelStateSemaphore;
 
-    public MLSyncUpCron(Client client, DiscoveryNodeHelper nodeHelper) {
+    public MLSyncUpCron(Client client, ClusterService clusterService, DiscoveryNodeHelper nodeHelper, MLIndicesHandler mlIndicesHandler) {
         this.client = client;
+        this.clusterService = clusterService;
         this.nodeHelper = nodeHelper;
+        this.mlIndicesHandler = mlIndicesHandler;
+        this.updateModelStateSemaphore = new Semaphore(1);
```

Comment:
should we add any fairness setting here?

Reply:
This is to control only one sync up job can update model state to avoid race condition.

Reply:
I saw Semaphore has another parameter `fair` :

    fair - true if this semaphore will guarantee first-in first-out granting of permits under contention, else false
    
  Let's say there's 3 sync up jobs (job1, job2, job3) and job1 is running. Don't we need to make sure that job2 should run after job1 is finished not job3?

Reply:
That's not necessary. We just need to make sure only 1 job updating the model state is fine. Job interval is relatively big enough (3s) that race condition is a rare case. If race condition happens, next job run will refresh model state again, no need to guarantee the first-in first-out.